lib/yt/http_request.rb

```diff
@@ -50,12 +50,24 @@ def run
           parse_response!
         end
       else
-        raise Yt::HTTPError, error_message
+        retry_run
       end
     end
 
   private
 
+    # retry the run method in case of a random 500 error from YouTube API
```

Comment:
@bruce-fs I agree, the retry should only happen in case the response is a ServerError.

In this PR, though, the retry happens every time, even when the response is 404.